.github/workflows/build_and_test_dbt.yaml

```diff
@@ -66,8 +66,12 @@ jobs:
         run: |
           if [[ $MODIFIED_RESOURCES_ONLY == 'true' ]]; then
             if [[ $MANUALLY_DISPATCHED == 'true' ]]; then
-              echo "Running build on manually selected resources"
-              dbt build -t "$TARGET" -s ${{ inputs.models }} --defer --state "$STATE_DIR" --resource-types model seed
+              echo "Running build on manually selected resources: $SELECTED_RESOURCES"
+              if [ "$SELECTED_RESOURCES" == "*" ]; then
+                dbt build -t "$TARGET" --resource-types model seed
+              else
+                dbt build -t "$TARGET" -s "$SELECTED_RESOURCES" --defer --state "$STATE_DIR" --resource-types model seed
+              fi
```

Comment:
See [this workflow run](https://github.com/ccao-data/data-architecture/actions/runs/9320458427/job/25657288808#step:8:41) for an example of this conditional working when `models == "*"` (canceled after confirming that 87 models had been selected, for the sake of time) and see [this workflow run](https://github.com/ccao-data/data-architecture/actions/runs/9320501875/job/25657425019#step:8:41) for a regression test demonstrating that it still works when `models == "default.vw_pin_sale default.vw_pin_address"`.

One semi-related thought I had while developing this: [I accidentally ran this workflow on `master` with `models == "*"`](https://github.com/ccao-data/data-architecture/actions/runs/9320431951) and was lucky enough to catch my mistake and cancel it before causing mass downtime. Should we gate the `*` operator so that it can't run when the target is `prod`?

Reply:
[Supposedly](https://docs.getdbt.com/reference/node-selection/syntax#specifying-resources) just using `ccao_data_athena` as the selector will build everything. I don't think it will be necessary to gate once we merge #462.

Reply:
**suggestion (blocking):** And to clarify, I don't see the need for a dedicated `*` selector when this is the case, so let's remove it unless I'm missing something.

Reply:
Makes sense, reverted in 76a88d1838a96642897832f0ca02cea57c8f12ae.